### PR TITLE
Add GHC 9.8.1-alpha3

### PR DESCRIPTION
**Pull request by bgamari**

### CODE REVIEW

File: ghcup-prereleases-0.0.7.yaml [hunk 1]
```diff
@@ -1060,13 +1060,129 @@ ghcupDownloads:
...
+          Darwin:
+            unknown_versioning:
+              dlHash: b2416c7f0379a95c10c15825d65bb7eea73061047fbd3c5a1fa0064d6befd90b
+              dlOutput: ghc-9.8.0.20230822-x86_64-darwin.tar.xz
+              dlSubdir: ghc-9.8.0.20230822-x86_64-apple-darwin
+              dlUri: https://downloads.haskell.org/~ghc/9.8.0.20230822/ghc-9.8.0.20230822-x86_64-apple-darwin.tar.xz
+          Linux_Alpine:
+            unknown_versioning:
+              dlHash: 7f040a364bf05c9b59e9a1bccdd07f41d0940185c1cda21e3448b94ed2eee712
+              dlOutput: ghc-9.8.0.20230822-x86_64-linux-alpine3_12.tar.xz
```
Comment (hasufell) [inline ghcup-prereleases-0.0.7.yaml:1095]: Static or dynamic?

**Comment by hasufell** [comment]
Please run the GHCup bindist pipeline

File: ghcup-prereleases-0.0.7.yaml (hunk 1 above)
Comment (bgamari) [reply ghcup-prereleases-0.0.7.yaml:1095]: Dynamic.

**Comment by bgamari** [comment]
> Please run the GHCup bindist pipeline

See https://gitlab.haskell.org/ghc/ghcup-ci/-/pipelines/83254.

**Comment by hasufell** [comment]
> > Please run the GHCup bindist pipeline
> 
> See https://gitlab.haskell.org/ghc/ghcup-ci/-/pipelines/83254.
> 
> I have updated the release checklist to ensure that this is done in the future.
> 

That is not the GHCup bindist pipeline.

GHCup is a release channel and separate from GHC and its pipeline. 

This separation is crucial. GHC gitlab pipeline results are insufficient evidence. GHCup has its own.

**Comment by bgamari** [comment]
> GHCup is a release channel and separate from GHC and its pipeline.

Note that I linked a GHCup pipeline, not a GHC pipeline. Nevertheless, I have started https://github.com/haskell/ghcup-metadata/actions/runs/5965101154/job/16181762818

**Comment by hasufell** [comment]
> Note that I linked a GHCup pipeline, not a GHC pipeline.

Yes, I'm aware. But GHCup should stay independent, including CI, testing, etc.

The separation between upstream and distributor is a very important one.

When dealing with an upstream bindist, we assume nothing. Upstream evidence is not enough. As such, a distribution channel is a quality gateway. We do want duplication of certain efforts here.

**Comment by hasufell** [comment]
> I have started https://github.com/haskell/ghcup-metadata/actions/runs/5965101154/job/16181762818. It appears to be failing for reasons unrelated to the metadata.

I see only one failure, which is expected: https://github.com/haskell/ghcup-metadata/actions/runs/5965101154/job/16181764098

Because armv7 bindists are not provided.

The other job is idle, because the mac M1 runner hosted by @angerman is offline. I've asked the Haskell Foundation multiple times about financial support for another runner, but got nothing so far. I won't be pushing that any further.

**Comment by hasufell** [comment]
> It appears to be failing for reasons unrelated to the metadata.

Well, it is related to the metadata: the bindist doesn't exist.

But it's not a blocker for releasing. 

The M1 job now succeded.

**Pull request merged**